packages/legacy/core/App/navigators/ContactStack.tsx

```diff
@@ -16,52 +16,79 @@ import WhatAreContacts from '../screens/WhatAreContacts'
 
 const ContactStack: React.FC = () => {
   const Stack = createStackNavigator<ContactStackParams>()
   const theme = useTheme()
   const { t } = useTranslation()
   const defaultStackOptions = useDefaultStackOptions(theme)
+  const [ContactListHeaderRignht] = useServices([TOKENS.COMPONENT_CONTACT_LIST_HEADER_RIGHT])
 
   return (
     <Stack.Navigator screenOptions={{ ...defaultStackOptions }}>
-      <Stack.Screen name={Screens.Contacts} component={ListContacts} options={{ title: t('Screens.Contacts') }} />
+      <Stack.Screen name={Screens.Contacts} component={ListContacts}
+        options={{ 
+          title: t('Screens.Contacts'),
+          headerRight: () => <ContactListHeaderRignht />,
+        }}
+      />
       <Stack.Screen
         name={Screens.ContactDetails}
         component={ContactDetails}
         options={{
           title: t('Screens.ContactDetails'),
+          headerRight: () => <ContactListHeaderRignht />,
         }}
       />
       <Stack.Screen
         name={Screens.RenameContact}
         component={RenameContact}
-        options={{ title: t('Screens.RenameContact') }}
+        options={() => ({ 
+          title: t('Screens.RenameContact'),
+          headerRight: () => <ContactListHeaderRignht />,
+      })}
+      />
+      <Stack.Screen name={Screens.Chat} component={Chat}
+        options={{ 
+          title: '',
+          headerRight: () => <ContactListHeaderRignht />,}}
+      />
+      <Stack.Screen name={Screens.WhatAreContacts} component={WhatAreContacts}
+        options={{ 
+          title: '',
+          headerRight: () => <ContactListHeaderRignht />,}}
       />
-      <Stack.Screen name={Screens.Chat} component={Chat} />
-      <Stack.Screen name={Screens.WhatAreContacts} component={WhatAreContacts} options={{ title: '' }} />
       <Stack.Screen
         name={Screens.CredentialDetails}
         component={CredentialDetails}
-        options={{ title: t('Screens.CredentialDetails') }}
+        options={() => ({ 
+          title: t('Screens.CredentialDetails'),
+          headerRight: () => <ContactListHeaderRignht />,
+      })}
       />
       <Stack.Screen
         name={Screens.CredentialOffer}
         component={CredentialOffer}
-        options={{ title: t('Screens.CredentialOffer') }}
+        options={() => ({ 
+          title: t('Screens.CredentialOffer'),
+          headerRight: () => <ContactListHeaderRignht />,
+        })}
       />
       <Stack.Screen
         name={Screens.ProofDetails}
         component={ProofDetails}
         options={() => ({
           title: '',
-          headerRight: () => <HeaderRightHome />,
+          headerRight: () => <ContactListHeaderRignht />,
         })}
       />
       <Stack.Screen
         name={Screens.ProofRequest}
         component={ProofRequest}
-        options={{ title: t('Screens.ProofRequest') }}
+        options={() => ({  title: t('Screens.ProofRequest'),
+        headerRight: () => <ContactListHeaderRignht />,
+      })}
```

Comment:
One major issue I see with this PR is that these header components are the same for many different screens. So if a Bifold user wants a different header component on a different screen in the same stack or if they want no header component for a certain screen in the same stack, this PR would prevent that.